go/vt/vttablet/tabletserver/health_streamer.go

```diff
@@ -61,9 +67,25 @@ type healthStreamer struct {
 	state   *querypb.StreamHealthResponse
 
 	history *history.History
+
+	ticks       *timer.Timer
+	dbConfig    dbconfigs.Connector
+	conns       *connpool.Pool
+	initSuccess bool
 }
 
 func newHealthStreamer(env tabletenv.Env, alias topodatapb.TabletAlias) *healthStreamer {
+	var newTimer *timer.Timer
+	var pool *connpool.Pool
+	if env.Config().SignalWhenSchemaChange {
+		reloadTime := env.Config().SchemaReloadIntervalSeconds.Get()
+		newTimer = timer.NewTimer(reloadTime)
```

Comment:
Should be `time.Duration(reloadTime) * time.Second` otherwise reload takes place every `30 * 60` nanoseconds.

Tip/opinion: when a variable denotes a time, and when it's not a `Duration` type, I always name it with the units. ie. I'd do:
```go
reloadTimeSeconds := env.Config().SchemaReloadIntervalSeconds.Get()
```
IMO this avoids much confusion.

Reply:
In this particular case, the returned object is a `time.Duration`, and not an integer as you might have thought. The duration is already in seconds. `SchemaReloadIntervalSeconds.Get()` looks like this:

```golang
return time.Duration(s * Seconds(1*time.Second))
```

Reply:
Oh, good. Thanks for explaining!